tests/integration/tls_tests/helpers.py

```diff
@@ -161,55 +158,58 @@ async def check_certs_correctly_distributed(
     Verifying certificates downloaded on the charm against the ones distributed by the TLS operator
     """
     app_name = app_name or await get_app_name(ops_test)
-    app_secret_id = await get_secret_id(ops_test, app_name)
     unit_secret_id = await get_secret_id(ops_test, unit.name)
-    app_secret_content = await get_secret_content(ops_test, app_secret_id)
     unit_secret_content = await get_secret_content(ops_test, unit_secret_id)
-    app_current_crt = app_secret_content["csr-secret"]
-    unit_current_crt = unit_secret_content["csr-secret"]
+
+    internal_unit_csr = unit_secret_content["int-csr-secret"]
+    external_unit_csr = unit_secret_content["ext-csr-secret"]
 
     # Get the values for certs from the relation, as provided by TLS Charm
     certificates_raw_data = await get_application_relation_data(
         ops_test, app_name, TLS_RELATION_NAME, "certificates"
     )
     certificates_data = json.loads(certificates_raw_data)
 
-    external_item = [
+    internal_item = [
         data
         for data in certificates_data
-        if data["certificate_signing_request"].rstrip() == unit_current_crt.rstrip()
+        if data["certificate_signing_request"].rstrip() == internal_unit_csr.rstrip()
     ][0]
-    internal_item = [
+    external_item = [
         data
         for data in certificates_data
-        if data["certificate_signing_request"].rstrip() == app_current_crt.rstrip()
+        if data["certificate_signing_request"].rstrip() == external_unit_csr.rstrip()
     ][0]
 
     # Get a local copy of the external cert
     external_copy_path = await scp_file_preserve_ctime(ops_test, unit.name, EXTERNAL_CERT_PATH)
 
     # Get the external cert value from the relation
-    relation_external_cert = "\n".join(external_item["chain"])
+    relation_external_cert = "\n".join(external_item["chain"]).strip()
 
     # CHECK: Compare if they are the same
     with open(external_copy_path) as f:
         external_contents_file = f.read()
-        assert relation_external_cert == external_contents_file
+        assert (
+            relation_external_cert == external_contents_file
+        ), f"Relation Content:\n{relation_external_cert}\nFile Content:\n{external_contents_file}\nMismatch."
 
     # Get a local copy of the internal cert
     internal_copy_path = await scp_file_preserve_ctime(ops_test, unit.name, INTERNAL_CERT_PATH)
```

Comment:
and then taking the same approach here

Reply:
I just grouped them together in a loop since they're identical